(docs) - Apply edits to automatically flagged errors (#1437)
FormidableLabs/urql: docs/advanced/authentication.md

The token is saved to storage that is persisted though sessions, e.g. `localStorage` on the web or `AsyncStorage` in React Native. The token is
added to each subsequent request in an auth header.

**Resume** — the user opens the application after having authenticated in the past. In this case, we should already have the token in persisted
storage. We fetch the token from storage and add to each request, usually as an auth header.

**Forced log out due to invalid token** — the user's session could become invalid for a variety reasons: their token expired, they requested to be
signed out of all devices, or their session was invalidated remotely. In this case, we would want to
also log them out in the application, so they
could have the opportunity to log in again. To do this, we want to clear any persisted storage, and redirect them to the application home or login page.

**User initiated log out** — when the user chooses to log out of the application, we usually send a logout request to the API, then clear any tokens
from persisted storage, and redirect them to the application home or login page.

> **Refresh (optional)** — this is not always implemented, but given that your API supports it, the
> user will receive both an auth token, and a refresh token,
> where the auth token is valid for a shorter duration of time (e.g. 1 week) than the refresh token

Reviewer: ```suggestion
**Refresh (optional)** — this is not always implemented; if your API supports it, the
user will receive both an auth token, and a refresh token. 
The auth token is valid for a shorter duration of time (e.g. 1 week) than the refresh token
```

When the application launches, the first thing we do is check whether the user has any auth tokens in persisted storage. This will tell us
whether to show the user the logged in or logged out view.

> The `isLoggedIn` prop should always be updated based on authentication state change e.g. set to
> `true` after the use has authenticated, and their tokens have been added to storage, and set to
> `false` if the user has been logged out, and their tokens have been cleared. It's important to clear
> or add tokens to a storage _before_ updating the prop in order for the auth exchange to work
> correctly.

Reviewer: ```suggestion
The `isLoggedIn` prop should always be updated based on authentication state change . E.g. set to
`true` after the user has authenticated and their tokens have been added to storage, and set to
`false` if the user has been logged out and their tokens have been cleared. It's important to clear
or add tokens to a storage _before_ updating the prop in order for the auth exchange to work
correctly.
```
